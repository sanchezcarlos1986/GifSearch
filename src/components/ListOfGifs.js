import React from 'react';
import PropTypes from 'prop-types';
import Gif from './Gif';

/**
 * Represents a list.
 * @constructor
 * @param {array} gifs - List of gifs
 */
export default function ListOfGifs({gifs}) {
  return (
    <div className="ListOfGifs">
      {gifs.map(({id, url, title}) => (
        <Gif key={id} id={id} url={url} title={title} />
      ))}
    </div>
  );
}

ListOfGifs.propTypes = {
  gifs: PropTypes.array.isRequired,
};
