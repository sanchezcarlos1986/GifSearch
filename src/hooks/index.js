export {default as useFetchGifs} from './useFetchGifs';
export {default as useForm} from './useForm';
