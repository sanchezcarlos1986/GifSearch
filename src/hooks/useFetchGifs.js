import {useState, useEffect} from 'react';
import {getGifs} from '~helpers';

/**
 * Represents a useEffect statement
 * @constructor
 * @param {string} category - Category to search and set data
 */
const useFetchGifs = category => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then(imgs =>
      setState({
        data: imgs,
        loading: false,
      }),
    );
  }, [category]);

  return state; // {data: [], loading: true}
};

export default useFetchGifs;
