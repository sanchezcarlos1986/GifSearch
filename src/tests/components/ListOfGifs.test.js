import React from 'react';
import '@testing-library/jest-dom';
import ListOfGifs from '../../components/ListOfGifs';
import {shallow} from 'enzyme';
import {useFetchGifs} from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs.js');

describe('ListOfGifs', () => {
  let props;
  let component;

  it('should renders ok', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    props = {
      params: {keyword: 'Mega Man'},
    };
    component = shallow(<ListOfGifs {...props} />);
    expect(component).not.toBeNull();
    expect(component.hasClass('ListOfGifs')).toBe(true);
    expect(component).toMatchSnapshot();
  });

  it('should renders items when images are loaded', () => {
    const gifs = [
      {
        id: 'abc',
        url: 'https://mygif.com',
        title: 'totoro',
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    props = {
      params: {keyword: 'Mega Man'},
    };
    component = shallow(<ListOfGifs {...props} />);

    expect(component).not.toBeNull();
    // expect(component.find('Gif').length).toBe(gifs.length);
    expect(component.find('Gif').prop('id')).toBe(gifs[0].id);
    expect(component.find('Gif').prop('url')).toBe(gifs[0].url);
    expect(component.find('Gif').prop('title')).toBe(gifs[0].title);
    expect(component).toMatchSnapshot();
  });
});
