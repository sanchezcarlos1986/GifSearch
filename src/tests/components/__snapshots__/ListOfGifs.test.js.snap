// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfGifs should renders items when images are loaded 1`] = `ShallowWrapper {}`;

exports[`ListOfGifs should renders ok 1`] = `ShallowWrapper {}`;
